### ISSUE TITLE
How will Bluetooth hacking work?

### ISSUE THREAD

**Issue by pahapoika** [cSploit/android#662]
So theres security features of Bluetooth.
1.Devices will not communicate unless you have paired the devices. And pairing is possible just if you know the BD_ADDR of other device.
2. It does hop the frequence like ehh rabbit?
Bluetooth uses AFH (Adaptive Frequence Hopping)
And it uses 79-channels in 2.4Ghz. And it hops between them in rate of 1600 per second.
3. It does encrypt the file that you are transfrening using your MAC adress 2-keys and random authenicating key. It does it using stream cipher called E0. Its 128-bit. 

You can brute force the BD_ADDR but that will take some time.

In Linux something like spoofing your victims computer to belive ur bluetooth keyboard are Its keyboards. But thats extremely hard.
I remember i had bluetooth hacking software on my Nokia on like year 2004. U were able to see contacts messanges and more. It were possible in bluetooth 1

**Comment by DeveloppSoft** [comment]
@pahapoika we may add tools like bluesnarfer or others.

**Comment by AlA777** [comment]
Yeah, i remember some app on SonyEricsson, that was pretty effective, when someone accept your pair invitation, already granted you full access to his telephone. Bluetooth was new type of communication, and everybody was excited, especially me. In 2005-6 i obtain my first Symbian Nokia N70,  and quick DSL connection.. things start changed rapidly..  

ok, sorry 4 OT..
Today i spend whole day by reading about Bluetooth, and practicing with bluesnarfer, spooftooph, etc. against my smartphones, tablets, tomtom, laptop, car radio, and neiberhood's tv and some reprostation. No success, but i don't have proper equipment.. :/ 
http://hakshop.myshopify.com/collections/wireless-gear
with regular bt client in mobile devices you can do only regular things. Same in laptop and dongle client in desktop, both Atheros chipset. 
https://drive.google.com/file/d/0B832qN0OMdvhOXJsVFoyX0htS0E/view?usp=drivesdk

**Comment by Megaeloelo** [comment]
That app was BTinfo but only works on Java mobile .

**Comment by lionliam96** [comment]
I think basic enumeration of surrounding bluetooth classic / le devices and services using android api would be a good place to start. I wrote  this dodgy test app a while back just to see what device services I could find (yes, I know it's terrible code but you get the basic idea):
https://github.com/lionliam96/Bluetooth-Scanner

Also, adding a database of bluetooth macs/service uuids could really help with researching bluetooth vulnerabilities. An implementation of the old obex push trick would be fun as well!

**Comment by DeveloppSoft** [comment]
Thanks @lionliam96 :smirk: I am having a look
